### PR TITLE
Allow visitors to post comments and view past comments

### PR DESCRIPTION
**Pull request by KerrynEer**
- Add form element in html
- use doPost() to send a post request to store user input
- Server stores an arraylist of comment objects (which contains text and the date it was created)

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -42,9 +43,23 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
     Gson gson = new Gson();
     String json = gson.toJson(messages);
     return json;
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userComment = request.getParameter("user-comment");
+
+    if (!userComment.isEmpty()) {
```
Comment (lingxz) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I think this check should be done on the frontend, i.e. if it's empty, don't even send a request

**Review by lingxz — COMMENTED** [review]
In general, it's better to split up the frontend and backend changes into different PRs. 

also, do you plan to add the datastore in a separate PR?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (KerrynEer) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Ah yes this will be better! Will make the change :)

**Comment by KerrynEer** [comment]
> In general, it's better to split up the frontend and backend changes into different PRs.
> 
> also, do you plan to add the datastore in a separate PR?

Ah alright, will keep this in mind for future PRs. Yup i will be creating a separate PR for the data store.

**Review by lingxz — APPROVED** [review]

**Pull request merged**